pkg/kubelet/kubelet_getters.go

```diff
@@ -310,8 +312,23 @@ func (kl *Kubelet) getPodVolumePathListFromDisk(podUID types.UID) ([]string, err
 		if err != nil {
 			return volumes, fmt.Errorf("could not read directory %s: %v", volumePluginPath, err)
 		}
-		for _, volumeDir := range volumeDirs {
-			volumes = append(volumes, filepath.Join(volumePluginPath, volumeDir))
+		unescapePluginName := utilstrings.UnescapeQualifiedName(volumePluginName)
+
+		if unescapePluginName != csi.CSIPluginName {
+			for _, volumeDir := range volumeDirs {
+				volumes = append(volumes, filepath.Join(volumePluginPath, volumeDir))
+			}
+		} else {
+			// For CSI volumes, the mounted volume path has an extra sub path "/mount", so also add it
+			// to the list if the mounted path exists.
+			for _, volumeDir := range volumeDirs {
+				path := filepath.Join(volumePluginPath, volumeDir)
+				volumes = append(volumes, path)
+				csimountpath := filepath.Join(path, "/mount")
```

Comment:
Can we reuse the logic in https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_mounter.go#L84? Maybe pull it out into a public function?

Reply:
Actually would it be possible to use mounter.GetPath() here for all plugins? Then we don't need plugin-specific logic.

Reply:
to get a new mounter, it requires volume spec, and pod id etc. It will need go through the volume spec reconstruction, which is not very easy... I think.

Reply:
Can we make a helper function out of https://github.com/kubernetes/kubernetes/blob/468af72a1547fe6c51c6f095206c1df981f8ae7a/pkg/volume/csi/csi_mounter.go#L84 that doesn't depend on mounter so that we don't have two places that define the path?

Reply:
where will the helper function be placed?
right now, the logic is depending on the plugin name, we define the path differently. You mean we have a helper function to contain this logic?

Reply:
ok, I made a helper function in csi_util.go. PTAL